helix-core/src/auto_pairs.rs

```diff
@@ -66,6 +63,36 @@ fn prev_char(doc: &Rope, pos: usize) -> Option<char> {
     doc.get_char(pos - 1)
 }
 
+/// calculate what the resulting range should be for an auto pair insertion
+fn get_next_range(
+    start_range: &Range,
+    offset: usize,
+    typed_char: char,
+    len_inserted: usize,
+) -> Range {
+    let end_head = start_range.head + offset + typed_char.len_utf8();
+
+    let end_anchor = match (start_range.len(), start_range.is_forward()) {
+        // if we have a zero width cursor, it shifts to the same number
+        (0, _) => end_head,
+
+        // if we are inserting for a regular one-width cursor, the anchor
+        // moves with the head
+        (1, true) => end_head - 1,
+        (1, false) => end_head + 1,
```

Comment:
I wonder if we can just merge the two because we already expand 0 width cursors to 1 in an after pass https://github.com/helix-editor/helix/blob/3e471e4bfecce11244579c1bd8a8e04ae83d9271/helix-core/src/selection.rs#L267-L273

I guess it could lose the direction so the current implementation is fine 👍🏻